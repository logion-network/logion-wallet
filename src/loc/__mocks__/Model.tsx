import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
