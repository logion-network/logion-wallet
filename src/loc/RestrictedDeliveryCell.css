.RestrictedDeliveryCell.checkbox-container {
    margin-top: 7px;
}

.RestrictedDeliveryCell.checkbox-container > * {
    width: 64px;
    margin-left: auto;
    margin-right: auto;
}
