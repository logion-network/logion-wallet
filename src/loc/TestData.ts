import { LegalOfficerCase, UUID, Hash } from "@logion/node-api";
import { LocData, HashString } from "@logion/client";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocData {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterLocId: loc.requesterLocId,
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId,
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            nature: HashString.fromValue(`File ${index}`),
            name: `File ${index}`,
            addedOn: "",
            published: false,
            restrictedDelivery: false,
            contentType: "text/plain",
            size: 42n,
            status: "DRAFT",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: HashString.fromValue(`Data ${index}`),
            value: HashString.fromValue(`Value ${index}`),
            addedOn: "",
            published: false,
            status: "DRAFT",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            nature: HashString.fromValue(`Nature ${index}`),
            addedOn: "",
            target: locFile.id.toString(),
            published: false,
            status: "DRAFT",
        })),
        closed: false,
        verifiedIssuer: false,
        issuers: [],
    };
}
