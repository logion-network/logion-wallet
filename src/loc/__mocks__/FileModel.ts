export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
