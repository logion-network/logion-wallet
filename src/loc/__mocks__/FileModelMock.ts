export let getAllDeliveries = jest.fn().mockResolvedValue({});
