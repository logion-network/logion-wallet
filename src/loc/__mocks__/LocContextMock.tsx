import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
