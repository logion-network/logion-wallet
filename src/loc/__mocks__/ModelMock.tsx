export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
