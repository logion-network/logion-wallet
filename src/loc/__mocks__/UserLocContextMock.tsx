import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
