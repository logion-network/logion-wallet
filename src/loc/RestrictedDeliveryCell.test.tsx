import { LocData } from "@logion/client";
import { UUID } from "@logion/node-api";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { DEFAULT_ADDRESS } from "src/common/TestData";
import { clickByName, shallowRender } from "src/tests"
import RestrictedDeliveryCell from "./RestricedDeliveryCell"
import { setLocRequest, setLocState } from "./__mocks__/LocContextMock";

jest.mock("./LocContext");

describe("RestrictedDeliveryCell", () => {

    it("renders", () => {
        const hash = "some-hash";
        setLocRequest({
            files: [{
                hash,
                name: "File name",
                nature: "Some nature",
                published: false,
                restrictedDelivery: false,
                submitter: DEFAULT_ADDRESS,
            }]
        } as unknown as LocData);
        const result = shallowRender(<RestrictedDeliveryCell hash={hash}/>);
        expect(result).toMatchSnapshot();
    });

    it("asks confirmation on authorization", () => testConfirm(false));
    it("asks confirmation on cancellation", () => testConfirm(true));

    it("does nothing on cancelled authorization", () => testDoesNothing(false));
    it("does nothing on cancelled cancellation", () => testDoesNothing(true));
});

const hash = "some-hash";

const expectedEndpoint = "/api/collection/b0944596-264d-4ebe-89e7-ab5562cbeff8/files/some-hash";

function renderGivenFile(restrictedDelivery: boolean) {
    const locData = {
        id: new UUID("b0944596-264d-4ebe-89e7-ab5562cbeff8"),
        files: [{
            hash,
            name: "File name",
            nature: "Some nature",
            published: false,
            restrictedDelivery,
            submitter: DEFAULT_ADDRESS,
        }]
    } as unknown as LocData;
    setLocRequest(locData);
    const put = jest.fn();
    const axios = { put };
    const client = {
        legalOfficers: [],
        buildAxios: () => axios,
    };
    setLocState({
        locsState: () => ({ client }),
        data: () => locData,
        refresh: () => {},
    });
    render(<RestrictedDeliveryCell hash={hash}/>);
    return put;
}

async function testConfirm(restrictedDelivery: boolean) {
    const put = renderGivenFile(restrictedDelivery);

    const checkbox = screen.getByRole("checkbox");
    await userEvent.click(checkbox);
    await clickByName("Confirm");

    expect(put).toBeCalledWith(expectedEndpoint, expect.objectContaining({
        restrictedDelivery: !restrictedDelivery,
    }));
}

async function testDoesNothing(restrictedDelivery: boolean) {
    const put = renderGivenFile(restrictedDelivery);

    const checkbox = screen.getByRole("checkbox");
    await userEvent.click(checkbox);
    await clickByName("Cancel");

    expect(put).not.toBeCalled();
}
