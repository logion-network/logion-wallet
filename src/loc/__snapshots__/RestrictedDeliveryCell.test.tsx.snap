// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestrictedDeliveryCell renders 1`] = `
<Cell
  content={
    <div
      className="RestrictedDeliveryCell checkbox-container"
    >
      <OverlayTrigger
        defaultShow={false}
        delay={500}
        overlay={
          <Tooltip
            id="RestrictedDeliveryCell-tooltip-some-hash"
            placement="right"
          >
            When the restricted delivery is activated, Collection-related NFT owners will be able to download a copy of the document.
          </Tooltip>
        }
        placement="bottom"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <div>
          <Checkbox
            checked={false}
            disabled={false}
            setChecked={[Function]}
            skin="Toggle black"
          />
        </div>
      </OverlayTrigger>
    </div>
  }
/>
`;
